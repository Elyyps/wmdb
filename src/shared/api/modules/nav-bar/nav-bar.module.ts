import { ILink } from "@app/core/link";

export interface INavBarModule {
  closeText: string;
  links: ILink[];
  logo: string;
  title: string;
}
