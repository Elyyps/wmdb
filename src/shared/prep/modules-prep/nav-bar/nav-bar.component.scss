.nav-bar {
  font-size: 14px;
  font-weight: 600;
  background: $global-background;
  padding: 13px 0;
  @media (max-width: $breakpoint-small-max) {
    padding: 10px 0;
  }
  &__holder {
    display: flex;
    justify-content: space-between;
    align-items: center;
    > div {
      display: flex;
      align-items: center;
    }
  }
  
  &__toggle {
    margin-left: 32px;
    @media (max-width: $breakpoint-small-max) {
      margin-left: 24px;
    }
  }
  
  &__links {
    padding: 0;
    margin: 0;
    display: flex;
    align-items: center;
    font-weight: 600;
    font-size: 16px;
    > li {
      padding-left: 30px;
    }
    a {
      color: $global-color;
      text-decoration: none;
      &:hover {
        color: $brand-primary;
      }
    }
  }
  
  .button--default {
    @media (max-width: $breakpoint-xsmall-max) {
      padding: 0;
      background: none;
      border: 0;
      line-height: 0;
      font-size: 0;
      .icon-svg {
        margin: 0;
        width: 17px !important;
        height: 17px !important;
      }
    }
  }
}