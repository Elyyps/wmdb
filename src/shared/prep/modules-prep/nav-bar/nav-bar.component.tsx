import * as React from "react";
import { Link } from "react-router-dom";
import "./nav-bar.component.scss";
import { LogoComponent } from "@app/prep/modules-prep/core";
import { ModalNavBar } from "@app/prep/modules-prep/core/modal-navbar";
import { HamburgerMenuComponent } from "@app/prep/modules-prep/hamburger-menu";
import MENU_ICON from "@assets/icons/menu.svg";

export interface INavBarComponentProps {}

const NavBarComponent = (props: INavBarComponentProps) => (
  <div className="nav-bar">
    <div className="uk-container">
      <div className="nav-bar__holder">
        <div>
          <LogoComponent />
        </div>

        <div>
          <ul className="nav-bar__links uk-visible@m">
            <li>
              <Link to={"#"}>Helpdesk</Link>
            </li>
            <li>
              <Link to={"#"}>Nieuwsbrief</Link>
            </li>
            <li>
              <Link to={"#"}>Mijn uitjes</Link>
            </li>
          </ul>

          <ModalNavBar
            className={"modal-menu nav-bar__toggle"}
            title={"Menu"}
            close={"Sluiten"}
            icon={MENU_ICON}
            iconSize="20px"
          >
            <HamburgerMenuComponent />
          </ModalNavBar>
        </div>
      </div>
    </div>
  </div>
);

export { NavBarComponent };
