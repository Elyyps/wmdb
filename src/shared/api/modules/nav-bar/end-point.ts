import { generateDummyNavbarData } from "./generate-dummy-data";
export const generateNavbarData = generateDummyNavbarData;
