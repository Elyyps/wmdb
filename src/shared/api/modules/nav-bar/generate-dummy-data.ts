import { INavBarModule } from "./nav-bar.module";
import Logo from "@assets/WMDB.svg";

export const generateDummyNavbarData = (): INavBarModule => ({
  closeText: "Sluiten",
  links: [
    { title: "Alle bedrijfsuitjes", url: "/organisaties" },
    { title: "Helpdesk", url: "/info-page" },
    { title: "Nieuwsbrief", url: "http://www.gooves.nl/", isExternalLink: true },
    { title: "Bekeken uitjes", url: "/overzicht" },
    { title: "Adverteren?", url: "/adverteren" }
  ],
  logo: Logo,
  title: "Menu",
  menuItems: [
    { title: "Alle bedrijfsuitjes", url: "/organisaties" },
    { title: "Helpdesk", url: "/info-page" },
    { title: "Nieuwsbrief", url: "http://www.gooves.nl/", isExternalLink: true },
    { title: "Bekeken uitjes", url: "/overzicht" },
    { title: "Adverteren?", url: "/adverteren" },
    { title: "Blog overview", url: "/blog-overview" }
  ]
});
