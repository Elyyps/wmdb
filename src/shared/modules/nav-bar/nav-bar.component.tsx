import * as React from "react";
import styles from "./nav-bar-component.module.scss";
import { LogoComponent } from "@app/core/logo/logo.component";
import { LinkComponent } from "@app/core/link";
import MENU_ICON from "@assets/icons/menu.svg";
import { INavBarModule } from "@app/api/modules/nav-bar/nav-bar.module";
import { NavBarModalComponent } from "../nav-bar-modal";
import { HamburgerMenuComponent } from "@app/core/hamburger-menu";

export interface INavBarComponentProps {
  navbarModule: INavBarModule;
}

const NavBarComponent = ({ navbarModule }: INavBarComponentProps) => (
  <div className={styles["nav-bar"]}>
    <div className="uk-container">
      <div className={styles["nav-bar__holder"]}>
        <div>
          <LogoComponent logo={navbarModule.logo} />
        </div>
        <div>
          <ul className={`${styles["nav-bar__links"]} uk-visible@m`}>
            {navbarModule.links.map((link, key) => (
              <li key={key}>
                <LinkComponent to={link.url}>{link.title}</LinkComponent>
              </li>
            ))}
          </ul>
          <NavBarModalComponent
            title={navbarModule.title}
            close={navbarModule.closeText}
            icon={MENU_ICON}
            iconSize="22px"
          >
            <HamburgerMenuComponent menuItems={navbarModule.menuItems} />
          </NavBarModalComponent>
        </div>
      </div>
    </div>
  </div>
);

export { NavBarComponent };
