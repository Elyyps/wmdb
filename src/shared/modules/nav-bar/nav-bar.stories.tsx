import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { NavBarComponent } from "./nav-bar.component";
import { generateNavbarData } from "@app/api/modules/nav-bar/end-point";

storiesOf("NavBar", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <NavBarComponent navbar={generateNavbarData()} />);
