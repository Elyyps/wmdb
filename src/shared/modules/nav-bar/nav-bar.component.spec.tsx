import React from "react";
import { shallow } from "enzyme";
import { NavBarComponent } from "./nav-bar.component";

describe("[NavBar]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<NavBarComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
