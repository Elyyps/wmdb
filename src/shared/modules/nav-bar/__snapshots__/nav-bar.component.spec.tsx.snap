// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[NavBar] should render component without crashing 1`] = `
<div
  className="nav-bar"
>
  <div
    className="uk-container"
  >
    <div
      className="nav-bar__holder"
    >
      <div>
        <LogoComponent
          logo="WMDB.svg"
        />
      </div>
      <div>
        <ul
          className="nav-bar__links uk-visible@m"
        >
          <li
            key="0"
          >
            <LinkComponent
              to="/info-page"
            >
              Helpdesk
            </LinkComponent>
          </li>
          <li
            key="1"
          >
            <LinkComponent
              to="/info-page"
            >
              Nieuwsbrief
            </LinkComponent>
          </li>
          <li
            key="2"
          >
            <LinkComponent
              to="/overzicht"
            >
              Mijn uitjes
            </LinkComponent>
          </li>
        </ul>
        <NavBarModalComponent
          close="Sluiten"
          icon="icons-menu.svg"
          iconSize="22px"
          title="Menu"
        >
          <HamburgerMenuComponent
            menuItems={
              Array [
                Object {
                  "title": "Helpdesk",
                  "url": "/info-page",
                },
                Object {
                  "title": "Nieuwsbrief",
                  "url": "/info-page",
                },
                Object {
                  "title": "Mijn uitjes",
                  "url": "/overzicht",
                },
              ]
            }
          />
        </NavBarModalComponent>
      </div>
    </div>
  </div>
</div>
`;
